dss-collection-nft/contracts/DSSCollection.cdc

```diff
@@ -0,0 +1,695 @@
+            }
+
+            return (&self.ownedNFTs[id] as &NonFungibleToken.NFT?)!
+        }
+
+        // borrowDSSCollectionNFT gets a reference to an NFT in the collection
+        //
+        pub fun borrowDSSCollectionNFT(id: UInt64): &DSSCollection.NFT? {
+            if self.ownedNFTs[id] != nil {
+                if let ref = &self.ownedNFTs[id] as auth &NonFungibleToken.NFT? {
+                    return ref! as! &DSSCollection.NFT
+                }
+                return nil
+            } else {
+                return nil
+            }
+        }
+
+        pub fun borrowViewResolver(id: UInt64): &AnyResource{MetadataViews.Resolver} {
+            let nft = (&self.ownedNFTs[id] as auth &NonFungibleToken.NFT?)!
+            let dssNFT = nft as! &DSSCollection.NFT
+            return dssNFT as &AnyResource{MetadataViews.Resolver}
+        }
+
+        destroy() {
+            destroy self.ownedNFTs
+        }
+
+        init() {
+            self.ownedNFTs <- {}
+        }
+    }
+
+    // public function that anyone can call to create a new empty collection
+    //
+    pub fun createEmptyCollection(): @NonFungibleToken.Collection {
+        return <- create Collection()
+    }
+
+    // An interface containing the Admin function that allows minting NFTs
+    //
+    pub resource interface NFTMinter {
+        pub fun mintNFT(collectionGroupID: UInt64, completedBy: String, level: UInt8): @DSSCollection.NFT
+    }
+
+    // A resource that allows managing metadata and minting NFTs
+    //
+    pub resource Admin: NFTMinter {
+
+        // Borrow a Collection Group
+        //
+        pub fun borrowCollectionGroup(id: UInt64): &DSSCollection.CollectionGroup {
+            pre {
+                DSSCollection.collectionGroupByID[id] != nil: "Cannot borrow collection group, no such id"
+            }
+
+            return (&DSSCollection.collectionGroupByID[id] as &DSSCollection.CollectionGroup?)!
+        }
+
+        // Borrow a Slot
+        //
+        pub fun borrowSlot(id: UInt64): &DSSCollection.Slot {
+            pre {
+                DSSCollection.slotByID[id] != nil: "Cannot borrow slot, no such id"
+            }
+
+            return (&DSSCollection.slotByID[id] as &DSSCollection.Slot?)!
+        }
+
+        // Borrow a Item
+        //
+        pub fun borrowItem(id: UInt64): &DSSCollection.Item {
+            pre {
+                DSSCollection.itemByID[id] != nil: "Cannot borrow item, no such id"
+            }
+
+            return (&DSSCollection.itemByID[id] as &DSSCollection.Item?)!
+        }
+
+        // Create a Collection Group
+        //
+        pub fun createCollectionGroup(
+            name: String,
+            description: String,
+            typeName: String,
+            startTime: UFix64?,
+            endTime: UFix64?,
+            timeBound: Bool
+        ): UInt64 {
+            let collectionGroup <- create DSSCollection.CollectionGroup(
```

Comment:
Yes, thought about this a bit. I don't think it would hurt to restrict name but not critical for our use case at the moment. Happy to add it if you feel strongly.